src/Expensify.js

```diff
@@ -129,9 +134,11 @@ class Expensify extends PureComponent {
         const previousAuthToken = lodashGet(prevProps, 'session.authToken', null);
         if (this.getAuthToken() && !previousAuthToken) {
             BootSplash.show({fade: true});
-            if (lodashGet(this.props, 'session.redirectToWorkspaceNewAfterSignIn', false)) {
-                Navigation.navigate(ROUTES.WORKSPACE_NEW);
-            }
+        }
+
+        if (this.getAuthToken() && this.props.betas && this.props.session.redirectToWorkspaceNewAfterSignIn) {
```

Comment:
How do you feel about using `lodashGet` to retrieve `betas` and `session.redirectToWorkspaceNewAfterSignIn`?

Reply:
Hmm I think it's unnecessary as this.props.session.redirectToWorkspaceNewAfterSignIn can only be boolean from onyx. If onyx doesn't pass this its false from default props so I don't see how it can be anything else and hence I don't feel like lodashGet is needed. 

Let me know if you feel there is some reason to have the lodashGet. I tested a few things around this and didn't see any console errors.

Reply:
> Hmm I think it's unnecessary as this.props.session.redirectToWorkspaceNewAfterSignIn can only be boolean from onyx

True, I was thinking it would be useful since we use `lodashGet` for getting the other Onyx data. I'd say do whatever feels right :D I see that the other `bool` onyx data aren't retrieved with `lodashGet` so I think you're right here 👍 

It's slightly different for `this.props.betas` since we're not looking at bools anymore, and `session.authToken` and `session.accountID` are defaulted to `null` both in `defaultProps` and using `lodashGet`, but I guess `lodashGet` is more useful b/c they're nested in the `session` prop. So I'm also ok with not using `lodashGet` for `this.props.betas` here if you don't want 😅

Reply:
Yeah i don't see any purpose for using lodashGet for betas or session.redirectToWorkspaceNewAfterSignIn. For betas the only time the code can fail is if this.props is undefined but that can't happen. For redirectToWorkspaceNewAfterSignIn the only time the code can fail is if this.props.session is undefined. Its definitely more possible for session to cause issues so I'm okay with updating session to use lodashGet. I'll update.

Reply:
Updated